libtsuba/src/RDGPartHeader.cpp

```diff
@@ -609,3 +670,79 @@ katana::to_json(json& j, const katana::PartitionTopologyMetadata& topomd) {
       {kPartitionTopologyMetadataEntriesKey, entries_vec},
   };
 }
+
+void
+katana::from_json(const nlohmann::json& j, katana::DynamicBitset& bitset) {
+  std::vector<uint64_t> vec;
+  size_t size;
+  j.at("bitset_vec").get_to(vec);
+  j.at("bitset_size").get_to(size);
+
+  for (const auto& val : vec) {
+    KATANA_LOG_WARN(
+        "EMCGINNIS from_json vec size = {}, val = {}", vec.size(), val);
+  }
+
+  bitset.clear();
+  bitset.resize(size);
+  for (const auto& val : vec) {
+    bitset.set(val);
+  }
+  for (const auto& val : vec) {
+    KATANA_LOG_VASSERT(bitset.test(val), "val {} unset!!", val);
+  }
+
+  KATANA_LOG_WARN(
+      "EMCGINNIS from_json vec size = {}, bitset_size = {}", vec.size(),
+      bitset.size());
+}
+
+void
+katana::to_json(nlohmann::json& j, const katana::DynamicBitset& bitset) {
+  std::vector<uint64_t> vec = bitset.GetOffsets<uint64_t>();
+  for (const auto& val : vec) {
+    KATANA_LOG_WARN(
+        "EMCGINNIS to_json vec size = {}, val = {}", vec.size(), val);
+  }
+  j = nlohmann::json{{"bitset_vec", vec}, {"bitset_size", bitset.size()}};
+  KATANA_LOG_WARN(
+      "EMCGINNIS to_json vec size = {}, bitset_size = {}", vec.size(),
+      bitset.size());
+}
+
+void
+katana::from_json(const nlohmann::json& j, katana::RDKLSHIndex& index) {
+  j.at("num_hashes_per_bucket").get_to(index.num_hashes_per_bucket_);
```

Comment:
Since we are going this route of storing arbitrary structures in our distributed graph we ought to at least version control the data structures so that we can fail the load if versions change

Reply:
I'm not sure I understand. This structure is already versioned by:
1) the RDG storage_format_version
2) the json key `kg.v1.rdk_lsh_index`

Any changes to this structure will require bumping the storage_format_version. We can also bump the version in the json key (all of the RDGPartHeader structures have this) but that hasn't been done historically.

Reply:
also, once this lands and is officially in storage format version 5, I'll update the input for https://github.com/KatanaGraph/katana/blob/master/python/test/test_storage_format.py which will then automatically detect changes to the data structure that are not accompanied by either the unstable flag or a format version bump.

Reply:
Why are we coupling the RDG storage format version with the version of the optional structures?

Reply:
The optional structures will, in most cases, get generated from the graphs they are attached to. This means that the optional structures are really only useful on a specific RDG, at a specific RDG storage format version. 
So it makes sense to extend the RDG storage format version to cover optional structures, which extends the protection from unintended changes.
This could be made more complicated later if we like.